app/forms/candidate_interface/volunteering_role_form.rb

```diff
@@ -3,16 +3,38 @@ class VolunteeringRoleForm
     include ActiveModel::Model
     include DateValidationHelper
 
-    attr_accessor :id, :role, :organisation, :details, :working_with_children,
-                  :start_date_day, :start_date_month, :start_date_year,
-                  :end_date_day, :end_date_month, :end_date_year
-
-    validates :role, :organisation, :working_with_children, presence: true
+    attr_accessor :id,
+                  :role,
+                  :organisation,
+                  :details,
+                  :working_with_children,
+                  :start_date_day,
+                  :start_date_month,
+                  :start_date_year,
+                  :start_date_unknown,
+                  :currently_working,
+                  :end_date_day,
+                  :end_date_month,
+                  :end_date_year,
+                  :end_date_unknown
+
+    validates :role,
+              :organisation,
+              :working_with_children,
+              presence: true
 
     validates :role, :organisation, length: { maximum: 60 }
-
     validates :start_date, date: { presence: true, future: true, month_and_year: true, before: :end_date }
-    validates :end_date, date: { future: true, month_and_year: true }, unless: :start_date_blank?
+
+    with_options if: :restructured_work_history_flag_active? do |form|
+      form.validates :start_date_unknown, inclusion: { in: %w[true false] }
+      form.validates :end_date_unknown, inclusion: { in: %w[true false] }
+      form.validates :currently_working, presence: true
+      form.validates :currently_working, inclusion: { in: %w[true false] }, if: -> { currently_working.present? }
+      form.validates :end_date, date: { presence: true, future: true, month_and_year: true }, if: -> { currently_working == 'false' }
```

Comment:
`currently_volunteering` as opposed to `currently_working`?

Reply:
Work experiences and volunteering experiences use the same DB table and this refers to a column `currently_working` so I think it's best we leave it as it is. 

The copy on the designs also says "Are you still working in this role?" for the view, so I think it's fine to stick with `currently_working` here too

Reply:
👍